### PR TITLE
Bradjohnl rem stag site

### PR DESCRIPTION
**Pull request by bradjohnl**
This PR implements the discussed changes:
- Remove the staging deployment.
- Start prod deployments directly when PRs are merged to dev
- Remove the PR-bot app and logic.

It also improves existing actions that are being migrated to Node v16 from v12.

### CODE REVIEW

**Comment by ipopescu** [comment]
@bradjohnl Looking at the patterns, they look ok to me, but I don't fully understand the changes. Should we get Tom to approve them? Or is there a way to test them?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -13,15 +11,15 @@ on:
...
+    if: github.ref == 'refs/heads/dev'
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v3
         with:
           ref: gh-pages
 
-      - uses: actions/upload-artifact@v2
+      - uses: actions/upload-artifact@v3
```
Comment (TomVasile) [inline .github/workflows/main.yml:22]: best practices from github on using actions is to pin to a specific sha hash

https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -36,7 +34,7 @@ jobs:
       - uses: actions/checkout@v3
 
       - name: Use Node.js ${{ env.selected_node_version }}
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
```
Comment (TomVasile) [inline .github/workflows/main.yml:37]: same deal

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -75,7 +73,7 @@ jobs:
         run: |
           echo "$prod_pages_fqdn" > ./docs/CNAME
 
-      - uses: actions/upload-artifact@v2
+      - uses: actions/upload-artifact@v3
```
Comment (TomVasile) [inline .github/workflows/main.yml:76]: same deal

**Review by TomVasile — CHANGES_REQUESTED** [review]
just some nit picks about using commit sha's

File: .github/workflows/main.yml (hunk 1 above)
Comment (TomVasile) [reply .github/workflows/main.yml:22]: fair, i'll approve

**Review by TomVasile — APPROVED** [review]

**Pull request merged**